code/math/vecmat.cpp

```diff
@@ -1150,47 +1150,66 @@ void find_point_on_line_nearest_skew_line(vec3d *dest, const vec3d *p1, const ve
 	vm_vec_scale_add(dest, p1, d1, numerator / denominator);
 }
 
-//make sure matrix is orthogonal
-//computes a matrix from one or more vectors. The forward vector is required,
-//with the other two being optional.  If both up & right vectors are passed,
-//the up vector is used.  If only the forward vector is passed, a bank of
-//zero is assumed
-//returns ptr to matrix
-void vm_orthogonalize_matrix(matrix *m_src)
-{
-	float umag, rmag;
-	matrix tempm;
-	matrix * m = &tempm;
-
-	vm_vec_copy_normalize(&m->vec.fvec,&m_src->vec.fvec);
 
-	umag = vm_vec_mag(&m_src->vec.uvec);
-	rmag = vm_vec_mag(&m_src->vec.rvec);
-	if (umag <= 0.0f) {  // no up vector to use..
-		if (rmag <= 0.0f) {  // no right vector either, so make something up
-			if (!m->vec.fvec.xyz.x && !m->vec.fvec.xyz.z && m->vec.fvec.xyz.y)  // vertical vector
-				vm_vec_make(&m->vec.uvec, 0.0f, 0.0f, 1.0f);
-			else
-				vm_vec_make(&m->vec.uvec, 0.0f, 1.0f, 0.0f);
+// normalizes only if above a small threshold, returns if normalized or not
+bool vm_maybe_normalize(vec3d* dst, const vec3d* src) {
+	float mag = vm_vec_mag(src);
+	if (mag < SMALL_NUM) return false;
```

Comment:
This should probably be `if ( abs(1.0f-mag) < std::numeric_limits<float>::epsilon() ) return false;`

Reply:
`SMALL_NUM` is used in several places in vecmat.cpp; I think that's acceptable.

Reply:
And incidentally, neither `abs` nor `1-mag` is the right thing to do here.